sql/parse/parse.go

```diff
@@ -1307,6 +1316,10 @@ func tableNameToUnresolvedTable(tableName sqlparser.TableName) *plan.UnresolvedT
 	return plan.NewUnresolvedTable(tableName.Name.String(), tableName.Qualifier.String())
 }
 
+func tableNameToUnresolvedTableAsOf(tableName sqlparser.TableName, asOf sql.Expression) *plan.UnresolvedTable {
```

Comment:
Anywhere else we should be calling this?

Reply:
Good question. I reviewed the usage again and most of the calls wouldn't need `as of` support (e.g. `alter table`, `drop table`). `show create table/view` does make sense though, so I went ahead and wired that up, even though the parser doesn't support specifying `as of` for those statements yet. When we add that to the parser, it'll start showing up in go-mysql-server and dolt and I can follow up with tests, too.